### PR TITLE
[Feature-8245][Alert] Add Alert Plugin Telegram

### LINKED ISSUE #8245: [Feature][Alert] Support Telegram Plugin
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Hi, All.

We want to add an alert plugin named Telegram for DolphinScheduler. And we would like to contribute to the community.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pinkhello**
## Purpose of the pull request

This pull request  add alert telegram plugin

## Brief change log

- *Add dolphinscheduler-alert-telegram to root `pom.xml`*
- *Add dolphinscheduler-alert-telegram to `dolphinscheduler-alert/dolphinscheduler-alert-plugins/pom.xml`*
- *Add `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram` folder*
- *Modify `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-slack/.../SlackAlertChannelFactory.java`*
- *Modify `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-slack/.../SlackParamsConstants.java`*
- *Modify `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-slack/.../SlackSender.java`*
- *Modify `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-slack/.../SlackSenderTest.java`*

## Verify this pull request

Added `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/.../TelegramAlertChannelFactoryTest.java`  and  `dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/.../TelegramSenderTest.java` to verify the change.

close #8245

### CODE REVIEW

**Comment by zhongjiajie** [comment]
@pinkhello  I approve and run CI. but please solved the conflict.

**Comment by zhongjiajie** [comment]
@pinkhello  I approve and run CI. but please solved the conflict.

**Comment by zhongjiajie** [comment]
Seem you missing the ASF license header

```
ERROR the following files don't have a valid license header: 
679
dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/README.MD
680
dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramAlertChannel.java
681
dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramAlertChannelFactory.java
682
dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramAlertConstants.java
683
dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramParamsConstants.java
684
dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramSender.java 
```

**Comment by pinkhello** [comment]
> @pinkhello I approve and run CI. but please solved the conflict. 

@zhongjiajie 
OK , DONE

**Comment by pinkhello** [comment]
> Seem you missing the ASF license header
done

**Comment by pinkhello** [comment]
> @pinkhello 我批准并运行 CI。但请解决冲突。

@zhongjiajie done

**Comment by pinkhello** [comment]
> Seem you missing the ASF license header
> 
> ```
> ERROR the following files don't have a valid license header: 
> 679
> dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/README.MD
> 680
> dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramAlertChannel.java
> 681
> dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramAlertChannelFactory.java
> 682
> dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramAlertConstants.java
> 683
> dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramParamsConstants.java
> 684
> dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramSender.java 
> ```

done

**Comment by pinkhello** [comment]
> @pinkhello I approve and run CI. but please solved the conflict.

@zhongjiajie resolved

**Comment by zhongjiajie** [comment]
Just approve to run the CI right now

**Comment by pinkhello** [comment]
> Just approve to run the CI right now

done

**Review by yimaixinchen — COMMENTED** [review]
errorCode might be standard.can replace error_code for errorCode

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramSender.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    }
+
+    private static class TelegramSendMsgResponse {
+
+        public boolean isOk() {
+            return null != ok && ok;
+        }
+
+        private Boolean ok;
+        private Integer error_code;
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-telegram/src/main/java/org/apache/dolphinscheduler/plugin/alert/telegram/TelegramSender.java:183]: Maybe use `errorCode` is better in current code style.

**Comment by pinkhello** [comment]
PasswordParam class check error！

**Comment by pinkhello** [comment]
> errorCode might be standard.can replace error_code for errorCode
done!  Sonar check error cause by PasswordParam

**Review by caishunfeng — APPROVED** [review]
LGTM

**Review by zhongjiajie — APPROVED** [review]
LGTM

**Comment by zhongjiajie** [comment]
Hi @pinkhello  , thanks for your contribution and welcome to join the community 🎉 . If you want to contribute but could not find issues, maybe you could start in #5689 or just search our issue list https://github.com/apache/dolphinscheduler/issues. Looking forward to your next contribution

**Comment by zhongjiajie** [comment]
BTW @pinkhello , we have WeChat groups to discuss dolphinscheduler’s feature and usage, and publish some tasks wanted some time. If you are interested in join WeChat group, please connect my WeChat id zhongjiajie955

**Pull request merged**